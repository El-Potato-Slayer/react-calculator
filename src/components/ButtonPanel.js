import PropTypes from 'prop-types';
import Button from './Button';

function ButtonPanel({ clickHandler }) {
  const buttonNames = [['AC', '+/-', '%', 'รท'],
    ['7', '8', '9', 'ร'],
    ['4', '5', '6', '-'],
    ['1', '2', '3', '+'],
    ['0', '.', '=']];

  function handleClick(buttonName) {
    return clickHandler(buttonName);
  }

  return (
    <div data-testid="button-panel" className="button-panel">
      {
        buttonNames.map((group) => (
          <div key={group}>
            {
              group.map((name) => (
                <Button key={name} name={name} clickHandler={() => handleClick(name)} />
              ))
            }
          </div>
        ))
      }
    </div>
  );
}

export default ButtonPanel;

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

ButtonPanel.defaultProps = {
  clickHandler: null,
};
