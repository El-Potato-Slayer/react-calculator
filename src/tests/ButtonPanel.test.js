import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('Renders ButtonPanel component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ButtonPanel />, div);
});

it('Renders ButtonPanel component correctly', () => {
  const { getByTestId } = render(
    <ButtonPanel />,
  );
  expect(getByTestId('button-panel')).toHaveTextContent('1');
});

it('Renders ButtonPanel component incorrectly', () => {
  const { getByTestId } = render(
    <ButtonPanel />,
  );
  expect(getByTestId('button-panel')).not.toHaveTextContent('abc');
});

it('matches snapshot', () => {
  const tree = renderer.create(<ButtonPanel />).toJSON();
  expect(tree).toMatchSnapshot();
});
