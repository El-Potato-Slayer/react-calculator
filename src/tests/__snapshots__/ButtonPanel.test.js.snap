// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button-panel"
  data-testid="button-panel"
>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      ร
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
